### DISCUSSION TITLE
How to implement skin-edit function by using filament attach-bone interface

### DISCUSSION THREAD

**Discussion by dandingol03** [sceneview/sceneview#94] [Q&A]
Hi.
&nbsp;Recently filament repo has release `attach-bone` interface, is there any possible for `SceneView` to using this attach-bone interface to implement skin-edit function.
&nbsp;Thanks for your time.

**Comment by onuralpszr** [comment]
Hello, 

https://github.com/SceneView/sceneview-android/blob/4b8a3b115a1cc5e8cf6453ba35fd4f2887160f5f/sceneview/src/main/java/com/google/ar/sceneform/rendering/RenderableInstance.java#L229

When you create renderableInstance you access to "getFilamentAsset" and towards you to "FilamentAsset" and then you can grab attachSkin / attach-bone section as well.

https://github.com/google/filament/blob/main/android/gltfio-android/src/main/java/com/google/android/filament/gltfio/FilamentAsset.java

**Comment by ThomasGorisse** [comment]
Bone manipulation (skeletal editing) is an advanced Filament feature. In v4.0.0-rc.1, you can access the Filament asset's skeleton and bones through the `ModelInstance`:

```kotlin
val model = rememberModelInstance(modelLoader, "models/character.glb")
// Access the underlying Filament asset for bone manipulation
val asset = model?.asset
// Filament's TransformManager can be used to modify individual bone transforms
```

However, high-level bone editing APIs are not exposed directly by SceneView. You'd need to work with Filament's `TransformManager` and `RenderableManager` directly to modify bone transforms for skin deformation. This is a low-level operation that requires understanding of Filament's entity/component system.
